catalog/data_share_details.go

```diff
@@ -0,0 +1,29 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceShareDetails() *schema.Resource {
```

Comment:
could you rename it to `DataSourceShareDetails` and `databricks_share` respectively? e.g. this is the expected pattern for data resources - same name as the resource, but only for reading